### ISSUE TITLE
[GSoC 2022 Work Summary] Select Tool and UI Enhancements Project

### ISSUE THREAD

**Issue by mohamedAdhamc** [synfig/synfig#3225]
This project consists of two main parts:

1. Select Tool: A tool that provides another way of transforming object in a similar manner to how lots of other graphics programs. [the current tool are vertex-editing tools]
2. Various UI Enhancements: This is a conglomerate of multiple features that aim to enhance the user interaction with Synfig.

The Features are all now implemented and are in various stages of the review process as follows:

**Select Tool:**
- Move Functionality: 
 #3036 status: in review
- Rotate Functionality: 
#3114 status:in review but requires merge of move functionality first.
~- Mirror Functionality: 
#3118 status: in progress.~ [No longer needed after select tool "redesign"]
- Scale Functionality: 
#3142 status: ready for review as well but also requires merge of previous two or just move functionality.
 
**Fixing UI lags/Issues:**
- Benchmarking/Testing various internal layer operations to see if caching optimzations make sense:
A- optimizing 'set_time()' : 
 #3149 

 **Various UI enhancements:**

- **A- Brush Tool Crashes Synfig:** 
 #3157 (1/2) 
- **B-  Crash on clicking on rows of Library Panel:**
 #3039  status: merged.
- **C- Crash when manipulating Duplicate Layer/ValueNode:** 
 #3180 #3182 
-  **Function to send to the top and bottom of the layer stack:** 
#3177 
- **Weird stutter in animation when Timeloop and Stroboscope Layers are used together:** 
Found a quick fix as described in the original issue but waiting on opinions on how it would be best addressed.

**Related PRs of things found along the way:**
- #3121 
- #3122 
- #3123 
- #3136 
- #3187 
- #3183 

Conclusion: This is the state of the various features at the date of completing the final gsoc submission. Most PRs are done but in various stages of the review process similar to last year. Further up to date updates on their status will be found in #3014
